heat/core/communication.py

```diff
@@ -460,6 +556,92 @@ def Iallgatherv(self, sendbuf, recvbuf, send_axis=0, recv_axis=None):
         return self.__allgather_like(self.handle.Iallgatherv, sendbuf, recvbuf, send_axis, recv_axis)
     Iallgatherv.__doc__ = MPI.Comm.Iallgatherv.__doc__
 
+    def __alltoall_w(self, sendbuf, recvbuf, send_axis, recv_axis, **kwargs):
+        """
+        Custom function for all-to-all operation if buffers are not continuous (i.e. not send_axis/recv_axis == 1/0
+
+        Parameters
+        ----------
+        sendbuf: Input send buffer; can be of type DNDarray, torch.Tensor, tuple = (torch.Tensor, send_counts, send_displ), or any other numpy supported type (only if send_axis == 0)
+        recvbuf: Input receive buffer; can be of type DNDarray, torch.Tensor, tuple = (torch.Tensor, send_counts, send_displ), or any other numpy supported type (only if send_axis == 0)
+        send_axis: future split axis, along which data blocks will be created that will be send to individual ranks
+                    if send_axis == recv_axis, an error will be thrown
+                    if send_axis or recv_axis are None, an error will be thrown
+        recv_axis: prior split axis, along which blocks are received from the individual ranks
+
+        Returns
+        -------
+        exit code: MPI4py alltoall_w
+        """
+        if (recv_axis == send_axis):
+            raise NotImplementedError('AllToAll for same axes not supported. Please choose send_axis and recv_axis to be different.')
+        if ( (send_axis == None) or (recv_axis == None) ):
+            raise NotImplementedError('AllToAll needs send_axis and recv_axis to be specified but was send_axis = {}, recv_axis = {}. Please set send_axis and recv_axis'.format(send_axis, recv_axis))
+
+        # ToDo: The option to explicitely specify the counts and displacements to be send still needs propper implementation
+        send_counts, send_displs, recv_counts, recv_displs = None, None, None, None,
```

Comment:
This will be a seperate issue, since it concerns also the general scatter_like funcitons (and the mpi_type_and_elements function)

Reply:
See Issue #371